src/lib/storage/dictionary_segment/attribute_vector_iterable.hpp

```diff
@@ -53,6 +53,9 @@ class AttributeVectorIterable : public PointAccessibleSegmentIterable<AttributeV
     explicit Iterator(const ValueID null_value_id, ZsIteratorType attribute_it, ChunkOffset chunk_offset)
         : _null_value_id{null_value_id}, _attribute_it{attribute_it}, _chunk_offset{chunk_offset} {}
 
+    // Only FixedSizeByteAligned is vectorizable, but we can't add IsVectorizable to its class.
```

Comment:
Uärgs, that means that if an innocent soul introduces another vector compression, they'll likely get compiler errors in the scan, because the decompression cannot be vectorized? 
If it is in any way possible to detect only the FixedSizeByteAlignedIterator-thing here, we should.

Reply:
Unfortunately not, because there is no `FixedSizeByteAlignedIterator` - it is just an iterator on an std::vector that we can't modify.

Reply:
But we can do something like

```c++
template<typename T> struct is_the_thing_we_are_looking_for { bool value = false; }
template<> struct is_the_thing_we_are_looking_for<the_thing_we_are_looking_for> { bool value = true; }
```

surely?

Reply:
I found a way of getting rid of this altogether